src/spacefinder/liveblog-adverts.ts

```diff
@@ -147,24 +145,60 @@ const stopListening = () => {
 	document.removeEventListener('liveblog:blocks-updated', onUpdate);
 };
 
-const fill = (rules: SpacefinderRules) => {
-	const options: SpacefinderOptions = { pass: 'inline1' };
+const getPreviousContentBlock = (element: HTMLElement): HTMLElement | null => {
+	const prevElement = element.previousSibling;
+	if (prevElement === null) return null;
+
+	if ((prevElement as HTMLElement).classList.contains('block')) {
+		return prevElement as HTMLElement;
+	}
+
+	return getPreviousContentBlock(prevElement as HTMLElement);
+};
+
+const insertMoreAds = () => {
+	const isMobile = getCurrentBreakpoint() === 'mobile';
+
+	let adContainerClass = '.ad-slot-container';
+	if (IS_SERVER_SIDE_MODE) {
+		adContainerClass += `.ad-slot-${isMobile ? 'mobile' : 'desktop'}`;
+	}
```

Comment:
Just checking, Are we saying in server-side mode the container class should be something like `.ad-slot-container.ad-slot-mobile`? This will successfully select the slot we require?

Reply:
Yep, that's correct. The containers exist for both sets of slots, whereas the slots will only exist for one set (mobile or not).

Reply:
Got it, thanks for confirming 🙂